config/confighttp/confighttp.go

```diff
@@ -242,12 +245,27 @@ func (hcs *ClientConfig) ToClient(ctx context.Context, host component.Host, sett
 	}
 
 	return &http.Client{
-		Transport: clientTransport,
-		Timeout:   hcs.Timeout,
-		Jar:       jar,
+		CheckRedirect: makeCheckRedirect(hcs.MaxRedirects),
+		Transport:     clientTransport,
+		Timeout:       hcs.Timeout,
+		Jar:           jar,
 	}, nil
 }
 
+// makeCheckRedirect checks if max redirects are exceeded
+func makeCheckRedirect(max *int) func(*http.Request, []*http.Request) error {
+	if max == nil {
+		return nil
+	}
+
+	return func(_ *http.Request, via []*http.Request) error {
+		if *max == (len(via) - 1) {
```

Comment:
Maybe a bit more intuitive. Let me know if you agree
```suggestion
		if len(via) > *max {
```